fmn/consumer/producer.py

```diff
@@ -55,7 +55,7 @@ def work(self):
         for pref in fmn.lib.models.Preference.list_batching(self.session):
 
             if not pref.context.name in self.backends:
-                return
+                continue
```

Comment:
Oh, and can you answer the "why is this needed" by adding a comment to the code? ☺